subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -1094,6 +1099,10 @@ uint8_t ull_adv_sync_pdu_set_clear(struct lll_adv_sync *lll_sync,
 	/* No TargetA in AUX_SYNC_IND */
 	/* No AdvA in AUX_SYNC_IND */
 
+	if (ter_com_hdr->ext_hdr_len) {
```

Comment:
Updated to explicit comparison. I didn't touch other places like https://github.com/zephyrproject-rtos/zephyr/blob/4e485c756525c148d2354810e8d84e443bd6217f/subsys/bluetooth/controller/ll_sw/ull_adv_sync.c#L1090
because there is similar function in ull_adv_aux.c: https://github.com/zephyrproject-rtos/zephyr/blob/4e485c756525c148d2354810e8d84e443bd6217f/subsys/bluetooth/controller/ll_sw/ull_adv_aux.c#L442 that has similar pattern.
In both cases the code should not pass MISRA check. It would be better to do it in separate PR where is clear information why the changes is done.